src/Alley/Validator/Comparison.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/*
+ * This file is part of the laminas-validator-extensions package.
+ *
+ * (c) Alley <info@alley.co>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Alley\Validator;
+
+use Laminas\Validator\Exception;
+
+final class Comparison extends BaseValidator
+{
+    private const SUPPORTED_OPERATORS = [
```

Comment:
Generally I'd advise adding PHPDoc for all properties and methods describing what they are for.

Reply:
While I agree it's unusual in the Alley context to omit them and would violate the coding standards for our WordPress work, PSR-12 itself doesn't require PHPDoc, and I've tried to use that as an opportunity to think about using it more carefully.

In the example of this particular repo:

- Nearly all of the methods and properties in these classes are implementations of an interface or of the Laminas abstract validator class. In these cases, I worry that redeclaring the documentation for the methods dilutes the interface and abstract class as sources of truth for that documentation, particularly the interfaces. (In this respect, PSR-12 and the like is better-suited to a class-heavy codebase than the WordPress standards are, IMO.)

- In that vein, I did add PHPDoc for the new method on the new abstract base class included in this repo: https://github.com/alleyinteractive/laminas-validator-extensions/blob/862a302a830c76b208e3f355d14cec9a135dd837/src/Alley/Validator/BaseValidator.php#L27-L33

- For the one-off properties on individual classes, the type declaration and variable name say pretty much exactly what the PHPDoc would say, e.g. `private ValidatorInterface $operatorOptionValidator`. I'm not sure what additional value the docs would provide, other than familiarity.

- For everything else, like constants that can't be typed, I tried to use descriptive variable names, again to try to make it so that the PHPDoc would be redundant.

Reply:
These are good arguments, and I support them.